pssh/ssh_client.py

```diff
@@ -100,6 +100,14 @@ def __init__(self, host,
                             else host)
         _user = host_config['user'] if 'user' in host_config else None
         user = user if user else _user
+
+        _port = int(host_config['port']) if 'port' in host_config else 22
+        port = port if port else _port
+
+        # FIXME: we take the first key
+        _pkey = paramiko.RSAKey.from_private_key_file(host_config['identityfile'][0]) if 'identityfile' in host_config else None
```

Comment:
Ok, ``paramiko`` [seems to manage](http://docs.paramiko.org/en/1.16/api/keys.html) ``RSA``, ``DSA`` and ``ECDSA`` keys. It's missing support for ``Ed25519`` algorithm (that is indicated from the ``openssh`` man page).

Weird enough ``paramiko`` doesn't guess the key type (should be easy as reading the first line I guess?) so the approach here could be one of

 1. guess by ourself reading the ``-----BEGIN <KEYTYPE> PRIVATE KEY-----``
 2. ``try/except`` for each type